### ISSUE TITLE
Support automatic creation of the MetalLB config when nodes are added/updated/deleted

### ISSUE THREAD

**Issue by surajssd** [packethost/packet-ccm#18]
Currently the only way to have a LoadBalancer functionality on Packet is using [MetalLB](https://metallb.universe.tf). It won't be an overstatement to say that the Kubernetes clusters running on Packet have to use MetalLB.

MetalLB is cool and works perfectly, the only thing that is error prone in MetalLB is that it needs a [configuration](https://metallb.universe.tf/configuration/#bgp-configuration). And sometimes folks forget to create/update the config as the newer nodes are added or removed from the cluster.

So I think this feature is a good candidate for the cloud controller manager to take care of.

**Task breakdown:**

- Accept a `configMap` which has information of the `EIPs` to use and `nodeSelectors`, it can look something like this:

```yaml
metallb:
  # List of all the EIP blocks you have purchased from Packet
  eips:
  - 147.23.89.56/28
  - 147.23.90.6/32
  nodeSelectors:
    # Nodes where the MetalLB pods should be running
    node-role.kubernetes.io/node: ""
```

- The controller watches node objects and creates/updates the metallb config in the metallb-system namespace. For this it will have to talk to the Packet API to fetch the information like what is the peer IP of the node in the private IP range.

- Also update the [existing secret](https://github.com/packethost/packet-ccm/tree/master/deploy/releases#create-config) config that is fed to the ccm pod to also take bgp Password, if it is set on the packet console.

```yaml
...
stringData:
  apiKey: "abc123abc123abc123"
  projectID: "abc123abc123abc123"
  bgpPassword: "abc123abc123abc123"
```

**Comment by deitch** [comment]
Hi @surajssd 

Yes, that is _exactly_ what we plan on doing; taking the excellent work @pereztr5 did [here](https://github.com/packet-labs/kubernetes-bgp/) and merging it in, so that the ccm will do it for you and it will "just work".

We actually planned on having it done by now, but some higher priority CSI work got in the way (a.k.a., "fix them bugs!" :-) ).

I am going to keep this open as a tracking issue, and so that you get updated as we progress through it.

**Comment by surajssd** [comment]
This will be awesome thing to have! I mean no pressure, but do you have any timeline on when this will happen?

**Comment by deitch** [comment]
Ha, no worries! Definitely targeted for this quarter.

**Comment by jacobsmith928** [comment]
@c0dyhi11 for you to watch as well!

**Comment by deitch** [comment]
Closed via #53 (thanks for pointing it out @displague )
